modules/partition-replicator/src/main/java/org/apache/ignite/internal/partition/replicator/raft/snapshot/outgoing/OutgoingSnapshotMetricsSource.java

```diff
@@ -0,0 +1,245 @@
+                            + " totalSnapshotInstallationTime={}ms, lastAppliedIndex={}, lastAppliedTerm={}, peers=[{}], learners=[{}],"
+                            + " catalogVersion={}]",
+                    partitionKey,
+                    snapshotId,
+                    h.rowsSent.value(),
+                    h.rowVersionsSent.value(),
+                    h.totalBytesSent.value(),
+                    h.outOfOrderRowsSent.value(),
+                    h.outOfOrderVersionsSent.value(),
+                    h.outOfOrderTotalBytesSent.value(),
+                    h.totalBatches.value(),
+                    h.avgBatchDuration.value(),
+                    h.minBatchDuration.value(),
+                    h.maxBatchDuration.value(),
+                    h.totalSnapshotInstallationTime.value(),
+                    h.lastAppliedIndex.value(),
+                    h.lastAppliedTerm.value(),
+                    h.peerList.value(),
+                    h.learnerList.value(),
+                    h.catalogVersion.value()
+            );
+        }
+    }
+
+    @Override
+    protected Holder createHolder() {
+        return new Holder();
+    }
+
+    class Holder implements AbstractMetricSource.Holder<Holder> {
+        final AtomicLongMetric rowsSent = new AtomicLongMetric("RowsSent", "Total rows sent");
+
+        final AtomicLongMetric rowVersionsSent = new AtomicLongMetric("RowVersionsSent", "Total row versions sent");
+
+        final AtomicLongMetric totalBytesSent = new AtomicLongMetric("TotalBytesSent", "Total bytes sent");
+
+        final AtomicLongMetric outOfOrderRowsSent = new AtomicLongMetric("OutOfOrderRowsSent", "Total out of order rows sent");
+
+        final AtomicLongMetric outOfOrderVersionsSent = new AtomicLongMetric("OutOfOrderVersionsSent", "Total out of order versions sent");
+
+        final AtomicLongMetric outOfOrderTotalBytesSent = new AtomicLongMetric("OutOfOrderTotalBytesSent", "Total out of order bytes sent");
+
+        final AtomicLongMetric totalBatches = new AtomicLongMetric("TotalBatches", "Total batches");
+
+        final AtomicLongMetric avgBatchDuration = new AtomicLongMetric("AvgBatchProcessingTime", "Average batch processing time");
+
+        final AtomicLongMetric minBatchDuration = new AtomicLongMetric("MinBatchProcessingTime", "Minimum batch processing time");
+
+        final AtomicLongMetric maxBatchDuration = new AtomicLongMetric("MaxBatchProcessingTime", "Maximum batch processing time");
+
+        final AtomicLongMetric totalSnapshotInstallationTime =
+                new AtomicLongMetric("TotalSnapshotInstallationTime", "Total snapshot installation time");
+
+        final AtomicLongMetric lastAppliedIndex = new AtomicLongMetric("LastAppliedIndex", "Last applied index");
+
+        final AtomicLongMetric lastAppliedTerm = new AtomicLongMetric("LastAppliedTerm", "Last applied term");
```

Comment:
In the code, we call it 'last applied index', but it seems more natural to talk about 'last included index' when we are talking about a Raft snapshot. Would it make sense to call it this way in the metrics (and maybe rename later the fields and methods in the code as well)?